File: GumiPresso_Server/src/main/java/com/ssafy/through/model/service/ProductService.java
package com.ssafy.through.model.service;

import java.util.List;

import com.ssafy.through.model.dto.Product;

public interface ProductService {

	Product select(String id);

	List<Product> search();
	
	List<Product> selectProductOrderByRating();
	List<Product> selectProductOrderByQuantity();
}

File: GumiPresso_User/app/src/main/java/com/ssafy/gumipresso/model/service/ProductService.kt
package com.ssafy.gumipresso.model.service

import com.ssafy.gumipresso.model.dto.Product
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface ProductService {
    @GET("/product/")
    suspend fun getProductList(): Response<List<Product>>

    @GET("/product/select/{product_id}")
    suspend fun getProduct(@Path("product_id") productId: String): Response<Product>

    @GET("/product/rating")
    suspend fun selectProductOrderByRating(): Response<List<Product>>

    @GET("/product/quantity")
    suspend fun selectProductOrderByQuantity(): Response<List<Product>>
}